From af7717c56364ef564ff45ab861099ec48cbcbff9 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 18 Mar 2025 09:30:40 +0100
Subject: [PATCH 814/954] feat(stm32mp2-fdts): add fixed IO compensation on
 vddio regulators for stm32mp231

IO compensation could alter duty cycle of high frequency output signal
on stm32mp25 SoC. To avoid such issue, fixed IO compensation is used.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 776935856d42cb6464860fadfaf3a63958afb731 feat(stm32mp2-fdts): add fixed IO compensation on vddio regulators for stm32mp231]
Change-Id: I4d054a5895e8829bef14df8db1dafc31af925663
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/444757
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp231.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/fdts/stm32mp231.dtsi b/fdts/stm32mp231.dtsi
index 8a8a4af4f7..cf7fc8c43e 100644
--- a/fdts/stm32mp231.dtsi
+++ b/fdts/stm32mp231.dtsi
@@ -361,22 +361,27 @@
 
 			vddio1: vddio1 {
 				regulator-name = "vddio1";
+				st,iocomp = <8 7>;
 			};
 
 			vddio2: vddio2 {
 				regulator-name = "vddio2";
+				st,iocomp = <8 7>;
 			};
 
 			vddio3: vddio3 {
 				regulator-name = "vddio3";
+				st,iocomp = <8 7>;
 			};
 
 			vddio4: vddio4 {
 				regulator-name = "vddio4";
+				st,iocomp = <8 7>;
 			};
 
 			vddio: vddio {
 				regulator-name = "vddio";
+				st,iocomp = <8 7>;
 			};
 		};
 
-- 
2.53.0

